packages/client/e2e/analytics.spec.ts

```diff
@@ -0,0 +1,145 @@
+
+      if (initiallastPositionAmount)
+        await expect(lastpositionLocator).not.toContainText(
+          initiallastPositionAmount,
+          { timeout: ElementTimeout.NORMAL },
+        )
+    })
+
+    test("Correct text color is displayed based on position value", async () => {
+      const initialpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const amount = Number(
+        (await initialpositionLocator.textContent())?.replace(/[.,]/g, ""),
+      )
+
+      if (amount < 0) {
+        expect(await initialpositionLocator.getAttribute("color")).toEqual(
+          "negative",
+        )
+      } else
+        expect(await initialpositionLocator.getAttribute("color")).toEqual(
+          "positive",
+        )
+    })
+
+    test.skip("Graph display coherent values", async () => {
+      // TODO
+      // more complex. may requires datatest-id to reliably assert graph content
+    })
+  })
+
+  test.describe("Positions section", () => {
+    test("Position nodes are showing tooltip information for each currencies", async () => {
+      for (const currency of currencies) {
+        const currencyCircle = analyticsPage
+          .locator("g.node")
+          .filter({ hasText: currency })
+          .first()
+        const currencyTooltip = analyticsPage
+          .locator("[data-testid='tooltip']", { hasText: currency })
+          .first()
+
+        await currencyCircle.hover()
+        await expect(currencyTooltip).toBeVisible({
+          timeout: ElementTimeout.AGGRESSIVE,
+        })
+        const regexp = RegExp(`${currency} [-+,.0-9]`, "g")
+        expect(
+          await currencyTooltip.textContent(),
+          `tooltip for ${currency} doesn't match expected pattern`,
+        ).toMatch(regexp)
+      }
+    })
+
+    test("Position nodes can be moved", async () => {
+      const nzdNode = analyticsPage
+        .locator("g.node")
+        .filter({ hasText: "NZD" })
+        .first()
+
+      const jpyNode = analyticsPage
+        .locator("g.node")
+        .filter({ hasText: "JPY" })
+        .first()
+
+      const nzdInitialPosition = await nzdNode.boundingBox()
+      await nzdNode.dragTo(jpyNode)
+      expect(nzdInitialPosition).not.toEqual(await nzdNode.boundingBox())
+    })
+  })
+
+  test.describe("PnL section", () => {
+    test("PnL value is displayed for each currencies", async () => {
+      const pnlSection = analyticsPage
+        .locator("div:last-of-type")
+        .filter({ hasText: "PnL" })
+        .last()
+
+      for (const currencypair of currencyPairs) {
+        const amountString = await pnlSection
+          .locator("div", {
+            hasText: currencypair,
+          })
+          .first()
+          .getByTestId("priceLabel")
+          .textContent()
+        const regexp = RegExp(`[-,.0-9km]`, "g")
+        expect(
+          amountString,
+          `amount for ${currencypair} doesn't match expected pattern`,
```

Comment:
minor: could do bit more accurate label for this - "expected" -> "abbreviated numerical"

Reply:
sure! I,ve updated the error msg as per suggested